core/common/scale.cpp

```diff
@@ -0,0 +1,506 @@
+      }
+    }  // namespace impl
+
+    namespace util {
+      template <class T>
+      auto makeSignReinterpreter(T value) {
+        return impl::SignReinterpreter<T>{value};
+      }
+    }  // namespace util
+
+    namespace fixedwidth {
+      template <class T>
+      T encodePlainInteger(T value) {
+        return impl::encodeLittleEndian(value);
+      }
+
+      template <class T>
+      T decodePlainInteger(T value) {
+        return impl::decodeLittleEndian(value);
+      }
+
+      uint8_t encodeTristate(Tristate value) {
+        switch (value) {
+          case Tristate::kNone:
+            return static_cast<uint8_t>(0x0);
+          case Tristate::kFalse:
+            return static_cast<uint8_t>(0x1);
+          case Tristate::kTrue:
+            return static_cast<uint8_t>(0x2);
+          default:
+            break;
+        }
+
+        assert(false && "Value is not Tristate");
+      }
+
+      expected::Result<Tristate, TristateDecodeError> decodeTristate(
+          uint8_t byte) {
+        switch (byte) {
+          case 0:
+            return expected::Value{Tristate::kNone};
+          case 1:
+            return expected::Value{Tristate::kFalse};
+          case 2:
+            return expected::Value{Tristate::kTrue};
+          default:
+            break;
+        }
+
+        return expected::Error{TristateDecodeError::kUnexpectedValue};
+      }
+
+      auto encodeInt8(int8_t value) {
+        return value;
+      }
+
+      auto encodeUInt8(uint8_t value) {
+        return value;
+      }
+
+      auto encodeInt16(uint16_t value) {
+        return encodePlainInteger(value);
+      }
+
+      auto encodeUint16(uint16_t value) {
+        return encodePlainInteger(value);
+      }
+
+      auto encodeInt32(uint32_t value) {
+        return encodePlainInteger(value);
+      }
+
+      auto encodeUint32(uint32_t value) {
+        return encodePlainInteger(value);
+      }
+
+      auto encodeInt64(int64_t value) {
+        return encodePlainInteger(value);
+      }
+
+      auto encodeUint64(uint64_t value) {
+        return encodePlainInteger(value);
+      }
+
+      auto decodeInt8(int8_t value) {
+        return value;
+      }
+
+      auto decodeUInt8(uint8_t value) {
+        return value;
+      }
+
+      auto decodeInt16(uint16_t value) {
+        return decodePlainInteger(value);
+      }
+
+      auto decodeUint16(uint16_t value) {
+        return decodePlainInteger(value);
+      }
+
+      auto decodeInt32(uint32_t value) {
+        return decodePlainInteger(value);
+      }
+
+      auto decodeUint32(uint32_t value) {
+        return decodePlainInteger(value);
+      }
+
+      auto decodeInt64(int64_t value) {
+        return decodePlainInteger(value);
+      }
+
+      auto decodeUint64(uint64_t value) {
+        return decodePlainInteger(value);
+      }
+
+      uint8_t encodeBool(bool value) {
+        return value ? static_cast<uint8_t>(0x01) : static_cast<uint8_t>(0x00);
```

Comment:
again redundant cast

Reply:
It seems to me, that these casts make the code look more human readable.

Reply:
Why? Type hint is already provided at function signature, and casting 0/1 to any number type looks suspicious

Reply:
Ok. Anyway already done.